common/src/main/java/com/wynntils/commands/WynntilsCommand.java

```diff
@@ -184,4 +467,39 @@ private static void addCommandDescription(
 
         text.append(clickComponent);
     }
+
+    private Object tryParseNewValue(Class<?> typeToParse, String value) {
```

Comment:
this could be replaced with `typeToParse.getConstructor(String.class).newInstance(value);` and the appropriate exception handling. you'd just need to define a new constructor in `CustomColor` - and it might be better to use hex codes for this instead? (i just realized after looking at the class that the hex parser always sets the alpha to 0, which is probably not what we want)

i feel like this method should also be moved to `StorageHolder` instead of here in the command class

Reply:
> this could be replaced with `typeToParse.getConstructor(String.class).newInstance(value);` and the appropriate exception handling. you'd just need to define a new constructor in `CustomColor` - and it might be better to use hex codes for this instead? (i just realized after looking at the class that the hex parser always sets the alpha to 0, which is probably not what we want)
> 
> 
> 
> i feel like this method should also be moved to `StorageHolder` instead of here in the command class

Agree with the constructor, but why StorageHolder, didn't we want to remove that? As for CustomColor, I will improve it's parsing in this PR.

Reply:
i think we should merge ConfigHolder and StorageHolder since there's little reason to have both - but for now you might as well add functionality to StorageHolder

Reply:
Regardless of placement, this could definitely benefit from the new functional switch expression in modern Java:

```
return switch (typeToParse) {
  Boolean.TYPE ->             if (value.equalsIgnoreCase("true") || value.equalsIgnoreCase("false")) {
                return Boolean.valueOf(value);
            } else {
                return null;
            };
 ...
default -> null;
}
```

or such (syntax might be slightly off).

Reply:
https://docs.oracle.com/en/java/javase/13/language/switch-expressions.html

Reply:
Switch doesn't work with `Class<?>`.